src/isEqual.ts

```diff
@@ -0,0 +1,47 @@
+import { Narrow } from './_narrow';
+import { purry } from './purry';
+
+/**
+ * Returns a function that determines if a value is _shallowly_ equal to the specified value using Object.is equality,
+ * and narrows the type of the value to match.
+ *
+ * @param specificValue the to compare to. This type should be narrower than the type being passed into the returned function.
+ * @returns a function that returns true if the the two values are equal.
+ * @signature
+ *    R.isEqual(specificValue)(wideValue);
+ * @example
+ *   [1, 2, 3].filter(R.isEqual(2)); // => [2]
+ * @dataLast
+ * @category Guard
+ */
+export function isEqual<WideValue, SpecificValue extends WideValue>(
+  specificValue: Narrow<SpecificValue>
```

Comment:
Are there specific cases where the Narrow utility is needed? If not, let's keep things simple here and not use this utility. I'm not sure there are real-life scenarios where it's useful.
```suggestion
  specificValue: SpecificValue
```

Reply:
This is specifically for the discriminated union case. Something like `hasPropSatisfying(union, 'type', isEqual('foo'))` is meaningless without a narrow type here.

Reply:
Added test cases that actually fail when this is missing.